tools/check-changelog-entry-file/main.go

```diff
@@ -53,12 +55,34 @@ func validateChangelog(filePath, body string) {
 	entry := changelog.Entry{
 		Body: body,
 	}
-	if err := entry.Validate(); err != nil {
-		log.Fatalf("Error validating changelog file: %s, err: %v", filePath, err)
+	notes := changelog.NotesFromEntry(entry)
+
+	if len(notes) < 1 {
+		log.Fatalf("Error validating changelog file: %s, no changelog entry found", filePath)
+	}
+
+	var unknownTypes []string
+	for _, note := range notes {
+		if !isValidType(note.Type) {
+			unknownTypes = append(unknownTypes, note.Type)
+		}
+	}
+	if len(unknownTypes) > 0 {
+		log.Fatalf("Error validating changelog file: %s. Unknown changelog types %v, please use only the configured changelog entry types %v", filePath, unknownTypes, allowedTypeValues)
 	}
+
 	fmt.Printf("Changelog entry file is valid: %s\n", filePath)
 }
 
+func isValidType(t string) bool {
```

Comment:
knit: a bit confusing this short variable names but i guess you got if as it is

Reply:
renamed to make more clear